### ISSUE TITLE
Remove deprecated IReactorUDP.connectUDP

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4075]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4075|
|Type|task|
|Created|2009-11-10 18:49:56Z|
|Branch|https://github.com/twisted/twisted/tree/remove-connect-udp-4075|

```IReactorUDP.connectUDP``` in [twisted/internet/interfaces.py](twisted/internet/interfaces.py) is marked as deprecated so let's get rid of it, it's probably been deprecated for ages..

<details><summary>Searchable metadata</summary>

```
trac-id__4075 4075
type__task task
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_remove_connect_udp_4075 branches/remove-connect-udp-4075
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1257878996000000 1257878996000000
changetime__1260128749000000 1260128749000000
version__None None
owner__ 
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [27630]) Branching to 'remove-connect-udp-4075'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra set owner to @exarkun|
|-|-|

Please merge after updating the copyright dates for:

 - [twisted/internet/udp.py](twisted/internet/udp.py)
 - [twisted/test/test_udp.py](twisted/test/test_udp.py)

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [27635]) Remove one more reference to connectUDP

refs [#4075](https://github.com/twisted/twisted/issues/4075)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun removed owner|
|-|-|

Oops, I found one more thing which needed to be removed.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

One pyflakes: `twisted/internet/udp.py:14: 'os' imported but unused`

+1!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [27686]) Merge remove-connect-udp-4075 

Author: exarkun
Reviewer: thijs, therve
Fixes: [#4075](https://github.com/twisted/twisted/issues/4075)

Remove the long deprecated IReactorUDP.connectUDP method.  The replacement is the
IUDPTransport.connect method, available to DatagramProtocols which are connected
to a UDP transport via IReactorUDP.listenUDP.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
